src/common/pf_bg_worker.c

```diff
@@ -0,0 +1,117 @@
+
+/*********************************************************************
+ *        _       _         _
+ *  _ __ | |_  _ | |  __ _ | |__   ___
+ * | '__|| __|(_)| | / _` || '_ \ / __|
+ * | |   | |_  _ | || (_| || |_) |\__ \
+ * |_|    \__|(_)|_| \__,_||_.__/ |___/
+ *
+ * www.rt-labs.com
+ * Copyright 2032 rt-labs AB, Sweden.
+ *
+ * This software is dual-licensed under GPLv3 and a commercial
+ * license. See the file LICENSE.md distributed with this software for
+ * full license information.
+ ********************************************************************/
+
+#include "pf_includes.h"
+
+#include <inttypes.h>
+
+/* Events handled by bg worker task */
+#define BG_JOB_EVENT_SAVE_NVM_DATA       BIT (0)
+#define BG_JOB_EVENT_UPDATE_PORTS_STATUS BIT (1)
+
+static void bg_worker_task (void * arg);
+static void run_save_file_job (pnet_t * net);
+static void run_update_ports_status_job (pnet_t * net);
+
+void pf_bg_worker_init (pnet_t * net)
+{
+   net->pf_bg_worker.events = os_event_create();
+   CC_ASSERT (net->pf_bg_worker.events != NULL);
+
+   os_thread_create (
+      "p-net_bg_worker",
+      10, // todo
+      4000,
+      bg_worker_task,
+      (void *)net);
+}
+
+int pf_bg_worker_run_job (pnet_t * net, pf_bg_job_t job_id)
+{
+   if (net == NULL)
+   {
+      printf ("pf_bg_worker_run_job net==NULL\n");
+      return -1;
+   }
+
+   switch (job_id)
+   {
+   case BG_JOB_SAVE_NVM_DATA:
+      os_event_set (net->pf_bg_worker.events, BG_JOB_EVENT_SAVE_NVM_DATA);
+      break;
+   case BG_JOB_UPDATE_PORTS_STATUS:
+      os_event_set (net->pf_bg_worker.events, BG_JOB_EVENT_UPDATE_PORTS_STATUS);
+      break;
+   default:
+      LOG_ERROR (
+         PNET_LOG,
+         "BGW(%d): Unsupported job %" PRIu32 "\n",
+         __LINE__,
+         job_id);
+      return -1;
+   }
+
+   return 0;
+}
+
+static void bg_worker_task (void * arg)
```

Comment:
```
/**
 * Task for running in a background thread
 *
 * @param arg              InOut: Thread argument, should be pnet_t *
 */
```

Reply:
OK

Reply:
I think there is still some static function without docstring.

Reply:
Now updated